iXBRLViewerPlugin/viewer/src/js/viewer.js

```diff
@@ -121,9 +121,14 @@ Viewer.prototype._preProcessiXBRL = function(n, docIndex, inHidden) {
             /* Is the element the only significant content within a <td> or <th> ? If
              * so, use that as the wrapper element. */
             node = $(n).closest("td,th").eq(0);
-            if (node.length == 1) {
-                var regex = "^[^0-9A-Za-z]*" + escapeRegex($(n).text()) + "[^0-9A-Za-z]*$";
-                if (node.text().match(regex) == null) {
+            const innerText = $(n).text();
+            if (node.length == 1 && innerText.length > 0) {
+                // Use indexOf rather than a single regex because innerText may
+                // be too long for the regex engine 
+                const outerText = $(node).text();
+                const start = outerText.indexOf(innerText);
+                const wrapper = outerText.substring(0, start) + outerText.substring(start + innerText.length);
+                if (wrapper.match('[0-9A-Za-z]') !== null) {
```

Comment:
This is fine, but a few thoughts:
1. If the concern is large cells, then concatenating the before/after (or even calling `.text()`) might be unwise; perhaps we should iterate the DOM children of td/th, skipping the element when it's found.  If multiple facts can be added per cell, perhaps the answer should be cached on the td/th to avoid redundantly processing large cells.
2. I wonder if `RegExp.test` is preferable to `String.match` with a string arg, which presumably requires recompiling the pattern.

Reply:
Regarding (1) I think there's quite a big difference between what's reasonable to search using a regex, and what it's reasonable to compile into a regex.  The string in question is around 34KB, and we have other places in the code where we handle it directly as it's the content of a tag.

(2) sounds like a reasonable improvement given that we are re-using exactly the same regex repeatedly.

Reply:
Oh, thank you for clarifying (1).  I indeed misread the code and didn't see that we were escaping+compiling the text into the regex.

Reply:
Switched to regex literal in af70b83..23388e1